astropy/io/fits/tests/test_diff.py

```diff
@@ -543,6 +573,52 @@ def test_partially_identical_files2(self):
             assert 'Data differs at [{}, 1]'.format(y + 1) in report
         assert '100 different pixels found (100.00% different).' in report
 
+    def test_partially_identical_files3(self):
+        """
+        Test files that have some identical HDUs but a different extension
+        name.
+        """
+
+        phdu = PrimaryHDU()
+        ehdu = ImageHDU(name='FOO')
+        hdula = HDUList([phdu, ehdu])
+        ehdu = BinTableHDU(name='BAR')
+        ehdu.header['EXTVER'] = 2
+        ehdu.header['EXTLEVEL'] = 3
+        hdulb = HDUList([phdu, ehdu])
+        diff = FITSDiff(hdula, hdulb)
+        assert not diff.identical
+
+        assert diff.diff_hdus[0][0] == 1
+
+        hdu_diff = diff.diff_hdus[0][1]
+        assert hdu_diff.diff_extension_types == ('IMAGE', 'BINTABLE')
+        assert hdu_diff.diff_extnames == ('FOO', 'BAR')
+        assert hdu_diff.diff_extvers == (1, 2)
+        assert hdu_diff.diff_extlevels == (1, 3)
+
+        report = diff.report()
+        assert 'Extension types differ' in report
+        assert 'a: IMAGE\n    b: BINTABLE' in report
+        assert 'Extension names differ' in report
+        assert 'a: FOO\n    b: BAR' in report
+        assert 'Extension versions differ' in report
+        assert 'a: 1\n    b: 2' in report
+        assert 'Extension levels differ' in report
+        assert 'a: 1\n    b: 2' in report
+
+    def test_diff_types(self):
+        phdu = PrimaryHDU()
+        ehdu = ImageHDU()
+        hdula = HDUList([phdu, ehdu])
+        ehdu = BinTableHDU()
+        hdulb = HDUList([phdu, ehdu])
+        diff = FITSDiff(hdula, hdulb)
+        assert not diff.identical
+        assert diff.diff_hdus[0][0] == 1
+
+
+
```

Comment:
Could you remove two of those blank lines? The other tests have one blank line between them.